public/js/admin.js

```diff
@@ -614,22 +617,21 @@ var updateVisibility = function(qid) {
                         id: qid
                     },
                     success: function(data) {
-                        displayQuestionTable();
                         // Toast notifiation
                         const msg = question['visible'] ? ' is now visible to the students' : ' is now&nbsp<u><b>not</b></u>&nbspvisible to the students';
 
-                        if(question['visible']){
-                            successSnackbar('Question ' + qid + msg);
+                        if (question['visible']) {
+                            successSnackbar('Question ' + questionNumber + msg);
                         } else {
-                            warningSnackbar('Question ' + qid + msg);
+                            warningSnackbar('Question ' + questionNumber + msg);
                         }
                     },
                     error: function(data) {
                         if (data['status'] === 401) {
                             window.location.href = '/';
                         } else {
-                            displayQuestionTable();
                             // Toast notification
+                            $('#checked-' + qid).prop('checked', !visibilityStatus);
```

Comment:
we can get rid of `visibilityStatus` and just use `question['visible']` like you have above

Reply:
good point